### ISSUE TITLE
[Documentation] WebClient.files.upload documentation is not up to date and is barely complete

### ISSUE THREAD

**Issue by baybal** [slackapi/node-slack-sdk#463]
### Description

[WebClient.files.upload documentation](http://slackapi.github.io/node-slack-sdk/reference/FilesFacet) is barely complete. It doesn't mention the existence of an example file and the approach to use in case the stream object lacks `options` properties that are needed by webclient (as revealed by @clavin in #307 ). It doesn't even mention that `file` input can be a ReadableStream compliant object.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [ ] enhancement (feature request)
- [ ] question
- [x] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

**Comment by aoberoi** [comment]
hi @baybal. I'm sorry we didn't meet your expectation on the quality of the documentation. I admit, the reference docs are a little lacking.

the example in the guide for `WebClient` is a little more useful: https://slackapi.github.io/node-slack-sdk/web_api#uploading-a-file. but specifically, i'm not sure what you expected in terms of mentioning an example file? isn't it obvious that if you're trying to upload a file, that you need a file? also, i've incorporated a fix in `3.15.0` so that you shouldn't need to use the fix that you mentioned from #307. I hope the guide's explanation is complete.

now that we've released v4, this issue might be a moot point. we're still supporting v3 for at least another 60 days, but i suggest moving your application to use v4 as soon as you can. the old docs should be back up after #486 is fixed. if there's anything you think we can specifically improve, i'm happy to take suggestions.

if you have any improvements in mind for the v4 docs, or any questions about using the file related methods, please open a new issue so we can triage these things separately.

**Comment by aoberoi** [comment]
The `v3` line of this project is no longer supported, so we're closing all issues that were specific to those releases. If this issue still persists in current releases, please open a new issue.
